src/MadLearning/MadLearning.API.Application/Events/Commands/CreateEvent.cs

```
@@ -24,9 +28,27 @@ public async Task<GetEventModelApiDto> Handle(CreateEvent request, CancellationT
                     request.dto.Location,
                     request.dto.Owner.ToPersonModel()!);
 
-            var createdEvent = await this.repository.CreateEvent(eventModel, cancellationToken);
+            try
+            {
+                var createdEvent = await this.repository.CreateEvent(eventModel, cancellationToken);
 
-            return createdEvent.ToApiDto();
+                this.calendarService.AddEvent(createdEvent);
+
+                return createdEvent.ToApiDto();
+            }
+            catch (StorageException)
+            {
+                this.logger.LogError("Could not create Event");
+
+                return null;
+            }
+            catch (CalendarException)
+            {
+                this.logger.LogError("Could not access Calendar");
+                await this.repository.DeleteEvent(eventModel.Id, cancellationToken);
+
+                return null;
+            }
```

Exceptions thrown should bubble up to the caller (controller action).
Do we have a clean way of informing users in UI what the error was about? So that when users have errors they report some error message they saw which we can correlate to code and logs.

---

I agree. But will not the controller respond with a logic status error code if when it is null? Since we are returning a specific object in the controller and not an IActionResult.

---

Returning null might end up in a valid status code (204 no content I'm guessing) , but that isn't helpful to the user. I think 500 is appropriate, and we should attach an error message that the UI can use to inform the user. 

Either way, in the application handler we shouldn't really care about how the producer of the command will handle a null, the application layer should just provide a meaningful error with as much context as needed. The producer of the command, in this case the API, will then deal with it in a way that makes sense for ASP.NET Core, since that happens to be the framework there. API and application layer are decoupled.

I think we should rethrow the exception here, since the cases of either the storage or Graph API being unavailable is an "exceptional" case.